From 7c4144c55610917b1a74ed6a27609e18468ddc8d Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 14 Dec 2017 01:11:18 +0800
Subject: [PATCH 3089/5242] MLK-17203 usb: host: xhci-plat: fix high bus freq
 release mismatch

commit  979482c70b6e2ecfe7421fcf2d03960687899bd2 from
https://source.codeaurora.org/external/imx/linux-imx.git

If the xhci platform device is already suspended, we can't
release high bus freq again, fix the high bus count mismatch.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/host/xhci-plat.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 6bc131d..42b699b 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -374,9 +374,11 @@ static int xhci_plat_remove(struct platform_device *dev)
 	clk_disable_unprepare(reg_clk);
 	usb_put_hcd(hcd);
 
+	if (!pm_runtime_suspended(&dev->dev))
+		release_bus_freq(BUS_FREQ_HIGH);
+
 	pm_runtime_set_suspended(&dev->dev);
 	pm_runtime_disable(&dev->dev);
-	release_bus_freq(BUS_FREQ_HIGH);
 
 	return 0;
 }
-- 
1.7.9.5

